packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -2486,7 +2486,7 @@ export function commitMutationEffects(
 
   setCurrentDebugFiberInDEV(finishedWork);
   commitMutationEffectsOnFiber(finishedWork, root, committedLanes);
-  setCurrentDebugFiberInDEV(finishedWork);
+  resetCurrentDebugFiberInDEV();
```

Comment:
This was an bug before and the layout effects just reset to previous so it left the last mutation fiber as the debug fiber after commit.